sdk/core/azure-core/src/main/java/com/azure/android/core/implementation/http/rest/RetrofitAPIClient.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.implementation.http.rest;
+
+import com.azure.android.core.util.serializer.JacksonAdapter;
+import com.azure.android.core.util.serializer.SerializerAdapter;
+import com.azure.android.core.util.serializer.SerializerEncoding;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.List;
+
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+
+/**
+ * Type to create Retrofit based service-interface implementation.
+ */
+public class RetrofitAPIClient {
+    /**
+     * Create a Retrofit based API client implementation for a given service-interface.
+     *
+     * Note: anuchan: This method takes minimal parameters needed to create service-interface
+     * implementation, when needed we can add overloads that takes custom
+     * SerializerAdapter, OkHttpClient, which is needed when we enable user to provide
+     * an existing OkHttpClient with SSL/Proxy/SocketPool etc.. configured.
```

Comment:
We need to eventually decide on right way to expose http configurations. For now starting with Ctr, we will circle-back once we build few clients (e.g. storage) on the top of this APIClient.